routers/api/v1/user/user.go

```diff
@@ -103,6 +103,10 @@ func GetInfo(ctx *context.APIContext) {
 	//     "$ref": "#/responses/notFound"
 
 	u := GetUserByParams(ctx)
+	if !u.IsVisibleToUser(ctx.User) {
+		ctx.JSON(http.StatusNotFound, nil)
```

Comment:
This must run after the `ctx.Written()` check below.
Also there still is an information leak because the 404 response differs between a "true" 404, and this case.
Try the following `.patch`:
```patch
diff --git a/routers/api/v1/user/helper.go b/routers/api/v1/user/helper.go
index fcdac257e..a3500e0ee 100644
--- a/routers/api/v1/user/helper.go
+++ b/routers/api/v1/user/helper.go
@@ -17,7 +17,7 @@ func GetUserByParamsName(ctx *context.APIContext, name string) *models.User {
 	user, err := models.GetUserByName(username)
 	if err != nil {
 		if models.IsErrUserNotExist(err) {
-			if redirectUserID, err := models.LookupUserRedirect(username); err == nil {
+			if redirectUserID, err2 := models.LookupUserRedirect(username); err2 == nil {
 				context.RedirectToUser(ctx.Context, username, redirectUserID)
 			} else {
 				ctx.NotFound("GetUserByName", err)
diff --git a/routers/api/v1/user/user.go b/routers/api/v1/user/user.go
index 44d392c47..1cf77bea4 100644
--- a/routers/api/v1/user/user.go
+++ b/routers/api/v1/user/user.go
@@ -103,11 +103,12 @@ func GetInfo(ctx *context.APIContext) {
 	//     "$ref": "#/responses/notFound"
 
 	u := GetUserByParams(ctx)
-	if !u.IsVisibleToUser(ctx.User) {
-		ctx.JSON(http.StatusNotFound, nil)
+	if ctx.Written() {
 		return
 	}
-	if ctx.Written() {
+	if !u.IsVisibleToUser(ctx.User) {
+		// fake ErrUserNotExist error message to not leak information about existence
+		ctx.NotFound("GetUserByName", models.ErrUserNotExist{Name: ctx.Params(":username")})
 		return
 	}
 
```

Reply:
Done